pkg/parquetquery/iters.go

```diff
@@ -602,11 +604,14 @@ func (t RowNumber) Preceding() RowNumber {
 // IteratorResult is a row of data with a row number and named columns of data.
 // Internally it has an unstructured list for efficient collection. The ToMap()
 // function can be used to make inspection easier.
+// WARNING: Be careful when adding slices of values from iterators to Entries, as
```

Comment:
I agree that it would be much safer to make `Entries` and `OtherEntries` private :+1: 

> likely too big for this PR

I think so too. I created a sub task in the [Add support for Arrays to TraceQL](https://github.com/grafana/tempo-squad/issues/435)